### ISSUE TITLE
terraform 0.12.16 and aci provide 0.1.2

### ISSUE THREAD

**Issue by francesco-supporlan** [ciscoecosystem/terraform-provider-aci#50]
Hi

When building a tf file with few resources (31 in my example file), at some point, creation is stuck and getting the message Still creating... [1m10s elapsed] forever.

Tried in macOS and CentOS and same issue.

Thanks for your help

**Comment by devarshishah3** [comment]
Hey, @francesco-supporlan Can you please attach your .tf files without secrets to the issue? I want to test the same in my dev environment to resolve this issue.

**Comment by francesco-supporlan** [comment]
Here are the 3 files used. They are generated using jinja2.



[Archive.zip](https://github.com/ciscoecosystem/terraform-provider-aci/files/3902071/Archive.zip)

**Comment by devarshishah3** [comment]
This is more of the Terraform + ACI issue rather than the provider issue. The root cause here is while creating a resource ACI returns the 405 error due to API rate limiting, Terraform doesn't find anything created and treats it as a null resource and fails to exit the process. There are lot of issues opened for this behavior on the Terraform repo and they are working on it. Meanwhile, I will try to find a workaround to resolve this issue and will update you.

**Comment by camrossi** [comment]
@devarshishah3  I do not think this is the case: 

I disabled rate limiting on my APIC and I still get the same 405 error. 

I checked a bit more in depth and I realise that apparently APIC always send back a 405 for every GET request coming from Terraform ACI plugin. (only terraform plugin)

Checking why I realised that all Terraforms GET request have a body length of 2 with a "{}" in the Payload. 

```
GET /api/node/mo/uni/vmmp-VMware/dom-ACI.json HTTP/1.1
Host: 10.67.185.106
User-Agent: Go-http-client/1.1
Content-Length: 2
Cookie: APIC-Request-Signature=XXX; APIC-Certificate-Algorithm=v1.0; APIC-Certificate-Fingerprint=fingerprint; APIC-Certificate-DN=uni/userext/user-automation/usercert-ansible
Accept-Encoding: gzip

{}  <== This is not supposed to be here! 
```

If you look at this code:
[MakeRestRequest](https://github.com/ciscoecosystem/aci-go-client/blob/master/client/client.go#L152)
The http.NewRequest method that we use in MakeRestRequest is always adding a body of {} even if the body is set to nil. I do not know Go at all so I am not sure if is an issue with the MakeRestRequest or http.NewRequest
